src/neo/Ledger/Blockchain.cs

```diff
@@ -356,12 +354,9 @@ protected override void OnReceive(object message)
                 case Block block:
                     OnInventory(block, false);
                     break;
-                case Transaction tx:
-                    OnTransaction(tx, true);
-                    break;
                 case Transaction[] transactions:
                     // This message comes from a mempool's revalidation, already relayed
-                    foreach (var tx in transactions) OnTransaction(tx, false);
+                    foreach (var tx in transactions) OnNewTransaction(tx);
```

Comment:
The policy changed. You need to re-verify the transactions. And if you revert this change, tps will be reduced to the same as before.

Reply:
I suppose Func `OnNewTransaction` will call func `VerifyStateDependent` and policy is checked there?

Reply:
Previously, if the policy was changed, the transactions needed to be completely re-verified, including `VerifyStateIndependent`. I'm not sure if it is unnecessary now.

Reply:
`VerifyStateIndependent` will check only the witness correctness now. Its price will be calculated in `VerifyDependent` together with others like transfer output, etc. Besides I observed that func `VerifyStateIndependent` is untouched in #2045. Is this correct?

Reply:
Can you fix this in another pr?

Reply:
#2107 should have solved this problem

Reply:
#2107 moves `TxRouter` to `NeoSystem` too. Can you create a pr for fixing `OnNewTransaction` only?